## Import Libraries and files

In [9]:
# Import libraries
import pandas as pd
import numpy as np
import os

In [11]:
# Path
path = r'/Users/michele/Desktop/CF/Files/Module6/AirBnb/Data'

In [729]:
# Import listings1
lst1 = pd.read_csv(os.path.join(path, 'Original', 'listings1.csv'))

In [731]:
# Import listings2
lst2 = pd.read_csv(os.path.join(path, 'Original', 'listings2.csv'))

In [733]:
# Import listings3
lst3 = pd.read_csv(os.path.join(path, 'Original', 'listings3.csv'))

In [735]:
# Import listings4
lst4 = pd.read_csv(os.path.join(path, 'Original', 'listings4.csv'))

In [737]:
# Import GeoJson file
geo = r'/Users/michele/Desktop/CF/files/Module6/AirBnb/Data/Original/bezirks.geojson'

## Data Cleaning

##### Lst1

In [29]:
# Lst1 size
lst1.shape

(13134, 75)

In [32]:
# Lst1 columns names
lst1.columns

Index(['id', 'listing_url', 'scrape_id', 'last_scraped', 'source', 'name',
       'description', 'neighborhood_overview', 'picture_url', 'host_id',
       'host_url', 'host_name', 'host_since', 'host_location', 'host_about',
       'host_response_time', 'host_response_rate', 'host_acceptance_rate',
       'host_is_superhost', 'host_thumbnail_url', 'host_picture_url',
       'host_neighbourhood', 'host_listings_count',
       'host_total_listings_count', 'host_verifications',
       'host_has_profile_pic', 'host_identity_verified', 'neighbourhood',
       'neighbourhood_cleansed', 'neighbourhood_group_cleansed', 'latitude',
       'longitude', 'property_type', 'room_type', 'accommodates', 'bathrooms',
       'bathrooms_text', 'bedrooms', 'beds', 'amenities', 'price',
       'minimum_nights', 'maximum_nights', 'minimum_minimum_nights',
       'maximum_minimum_nights', 'minimum_maximum_nights',
       'maximum_maximum_nights', 'minimum_nights_avg_ntm',
       'maximum_nights_avg_ntm', 'ca

In [35]:
# Lst1 preview
lst1.head()

id                         listing_url       scrape_id last_scraped  \
0   3176   https://www.airbnb.com/rooms/3176  20230916061000   2023-09-16   
1   9991   https://www.airbnb.com/rooms/9991  20230916061000   2023-09-16   
2  14325  https://www.airbnb.com/rooms/14325  20230916061000   2023-09-16   
3  16644  https://www.airbnb.com/rooms/16644  20230916061000   2023-09-16   
4  17904  https://www.airbnb.com/rooms/17904  20230916061000   2023-09-16   

        source                                               name  \
0  city scrape  Rental unit in Berlin · ★4.63 · 1 bedroom · 2 ...   
1  city scrape  Rental unit in Berlin · ★5.0 · 4 bedrooms · 7 ...   
2  city scrape  Rental unit in Berlin · ★4.67 · Studio · 1 bed...   
3  city scrape  Condo in Berlin · ★4.72 · 1 bedroom · 1 bed · ...   
4  city scrape  Rental unit in Berlin · ★4.77 · Studio · 1 bed...   

                                         description  \
0  This beautiful first floor apartment  is situa...   
1  4 bedroom with very large windows and outstand...   
2  The apartment is located on the upper second f...   
3  Light and sunny 2-Room-turn of the century-fla...   
4  - apt is available starting April 1, 2023<br /...   

                               neighborhood_overview  \
0  The neighbourhood is famous for its variety of...   
1  Prenzlauer Berg is an amazing neighbourhood wh...   
2                                                NaN   
3  Our Part of Kreuzberg is just the best. Good v...   
4                                                NaN   

                                         picture_url  host_id  ...  \
0  https://a0.muscache.com/pictures/243355/84afcf...     3718  ...   
1  https://a0.muscache.com/pictures/42799131/59c8...    33852  ...   
2  https://a0.muscache.com/pictures/508703/24988a...    55531  ...   
3  https://a0.muscache.com/pictures/73759174/e2ef...    64696  ...   
4  https://a0.muscache.com/pictures/d9a6f8be-54b9...    68997  ...   

  review_scores_communication review_scores_location review_scores_value  \
0                        4.69                   4.92                4.62   
1                        5.00                   4.86                4.86   
2                        4.84                   4.58                4.42   
3                        4.86                   4.67                4.71   
4                        4.92                   4.87                4.65   

                                             license instant_bookable  \
0  First name and Last name: Nicolas Krotz <br/> ...                f   
1                                  03/Z/RA/003410-18                f   
2                                                NaN                f   
3                                                NaN                f   
4                                                NaN                f   

  calculated_host_listings_count calculated_host_listings_count_entire_homes  \
0                              1                                           1   
1                              1                                           1   
2                              4                                           4   
3                              2                                           2   
4                              1                                           1   

  calculated_host_listings_count_private_rooms  \
0                                            0   
1                                            0   
2                                            0   
3                                            0   
4                                            0   

  calculated_host_listings_count_shared_rooms reviews_per_month  
0                                           0              0.85  
1                                           0              0.08  
2                                           0              0.16  
3                                           0              0.30  
4                        

In [739]:
# Select only relevant variables
listing1 = lst1[['id', 'last_scraped', 'host_id','host_location', 'neighbourhood_group_cleansed', 'latitude','longitude', 'room_type', 'accommodates','bathrooms_text', 'bedrooms', 'beds', 'price','minimum_nights', 'maximum_nights', 'availability_365', 'license', 'instant_bookable','calculated_host_listings_count']]

In [425]:
# Look for duplicates
dup = listing1.duplicated()
dup.sum()

0

In [427]:
# Check for missing values 
listing1.isnull().sum()

id                                   0
last_scraped                         0
host_id                              0
host_location                     2288
neighbourhood_group_cleansed         0
latitude                             0
longitude                            0
room_type                            0
accommodates                         0
bathrooms_text                       6
bedrooms                          4874
beds                               162
price                                0
minimum_nights                       0
maximum_nights                       0
availability_365                     0
license                           5122
instant_bookable                     0
calculated_host_listings_count       0
dtype: int64

In [741]:
# Keep only numerical value for bathroom text
listing1 = listing1.assign(bathrooms=listing1['bathrooms_text'].str.extract(r'(\d+[\.,]?\d*)'))

In [743]:
# Change data type for baths col.
listing1['bathrooms'] = listing1['bathrooms'].astype(float)

In [745]:
# Drop bathrooms_text col.
listing1.drop('bathrooms_text', axis=1, inplace=True)

In [747]:
# Check for missing values in bathrooms
listing1['bathrooms'].isnull().sum()

41

In [37]:
# Summary Statistics
listing1.describe()

id       host_id      latitude     longitude  accommodates  \
count  1.313400e+04  1.313400e+04  13134.000000  13134.000000  13134.000000   
mean   2.394956e+17  1.443509e+08     52.509939     13.401692      2.917314   
std    3.722776e+17  1.635086e+08      0.033458      0.067874      1.853629   
min    3.176000e+03  1.581000e+03     52.369040     13.107580      1.000000   
25%    1.651798e+07  1.374351e+07     52.490353     13.363094      2.000000   
50%    3.742721e+07  5.969585e+07     52.509770     13.411290      2.000000   
75%    6.504403e+17  2.509105e+08     52.532688     13.437867      4.000000   
max    9.811597e+17  5.369634e+08     52.656110     13.721390     16.000000   

          bedrooms          beds  minimum_nights  maximum_nights  \
count  8260.000000  12972.000000    13134.000000    13134.000000   
mean      1.484140      1.824237       43.555505      640.684940   
std       0.814338      1.419906       54.080779      492.657922   
min       1.000000      1.000000        1.000000        1.000000   
25%       1.000000      1.000000        2.000000      123.500000   
50%       1.000000      1.000000        7.000000      365.000000   
75%       2.000000      2.000000       92.000000     1125.000000   
max      14.000000     22.000000     1100.000000     9999.000000   

       availability_365  calculated_host_listings_count     bathrooms  
count      13134.000000                    13134.000000  13093.000000  
mean         126.761840                        9.252627      1.133736  
std          135.373208                       27.820488      0.397568  
min            0.000000                        1.000000      0.000000  
25%            0.000000                        1.000000      1.000000  
50%           73.000000                        1.000000      1.000000  
75%          262.000000                        3.000000      1.000000  
max          365.000000                      187.000000      8.500000

In [749]:
# Imputing mean for 'bathrooms', 'bedrooms', 'beds'
listing1['bathrooms'] = listing1['bathrooms'].fillna(value=1)
listing1['bedrooms'] = listing1['bedrooms'].fillna(value=1)
listing1['beds'] = listing1['beds'].fillna(value=2)

In [751]:
# Add boolean col for licence
listing1['license_bool'] = listing1['license'].notnull()

In [753]:
# Drop license col.
listing1.drop(columns='license', axis=1, inplace=True)

In [163]:
# Check data types 
listing1.dtypes

id                                  int64
last_scraped                       object
host_id                             int64
host_location                      object
neighbourhood_group_cleansed       object
latitude                          float64
longitude                         float64
room_type                          object
accommodates                        int64
bedrooms                          float64
beds                              float64
price                              object
minimum_nights                      int64
maximum_nights                      int64
availability_365                    int64
instant_bookable                   object
calculated_host_listings_count      int64
bathrooms                         float64
license_bool                         bool
dtype: object

In [453]:
# Check price that should be float64 instead of obj.
print(listing1['price'])

0         $83.00
1        $180.00
2         $70.00
3         $90.00
4         $25.00
          ...   
13129    $100.00
13130     $90.00
13131     $95.00
13132    $130.00
13133     $71.00
Name: price, Length: 13134, dtype: object


In [755]:
# Replace the $ sign in price
listing1['price'] = listing1['price'].str.replace('$', '', regex=False).str.replace(',', '', regex=False).str.strip()

In [757]:
# Change data type for price
listing1['price'] = listing1['price'].astype(float)

In [459]:
# Summary statistics for price
listing1['price'].describe()

count    13134.000000
mean       121.873078
std        404.656643
min          8.000000
25%         50.000000
50%         84.000000
75%        142.000000
max      24999.000000
Name: price, dtype: float64

In [759]:
# Change data type for 'id', 'host_id', 'host_is_superhost', 'instant_bookable'
listing1 = listing1.astype({'id':str, 'host_id':str, 'instant_bookable':bool})

In [761]:
# Change columns names dict.
new_name = {'neighbourhood_group_cleansed':'neighbourhood', 'license_bool':'license', 'calculated_host_listings_count':'host_listings' }

In [763]:
# Apply dictionary
listing1.rename(columns=new_name, inplace=True)

In [467]:
# Check the changes
listing1.columns

Index(['id', 'last_scraped', 'host_id', 'host_location', 'neighbourhood',
       'latitude', 'longitude', 'room_type', 'accommodates', 'bedrooms',
       'beds', 'price', 'minimum_nights', 'maximum_nights', 'availability_365',
       'instant_bookable', 'host_listings', 'bathrooms', 'license'],
      dtype='object')

##### Lst2

In [63]:
# Lst2 size
lst2.shape

(13327, 75)

In [65]:
# Lst2 columns names
lst2.columns

Index(['id', 'listing_url', 'scrape_id', 'last_scraped', 'source', 'name',
       'description', 'neighborhood_overview', 'picture_url', 'host_id',
       'host_url', 'host_name', 'host_since', 'host_location', 'host_about',
       'host_response_time', 'host_response_rate', 'host_acceptance_rate',
       'host_is_superhost', 'host_thumbnail_url', 'host_picture_url',
       'host_neighbourhood', 'host_listings_count',
       'host_total_listings_count', 'host_verifications',
       'host_has_profile_pic', 'host_identity_verified', 'neighbourhood',
       'neighbourhood_cleansed', 'neighbourhood_group_cleansed', 'latitude',
       'longitude', 'property_type', 'room_type', 'accommodates', 'bathrooms',
       'bathrooms_text', 'bedrooms', 'beds', 'amenities', 'price',
       'minimum_nights', 'maximum_nights', 'minimum_minimum_nights',
       'maximum_minimum_nights', 'minimum_maximum_nights',
       'maximum_maximum_nights', 'minimum_nights_avg_ntm',
       'maximum_nights_avg_ntm', 'ca

In [765]:
# Select only relevant variables
listing2 = lst2[['id', 'last_scraped', 'host_id','host_location', 'neighbourhood_group_cleansed', 'latitude','longitude', 'room_type', 'accommodates','bathrooms_text', 'bedrooms', 'beds', 'price','minimum_nights', 'maximum_nights', 'availability_365', 'license', 'instant_bookable','calculated_host_listings_count']]

In [471]:
# Look for duplicates
dup = listing2.duplicated()
dup.sum()

0

In [473]:
# Look for missing values
listing2.isnull().sum()

id                                    0
last_scraped                          0
host_id                               0
host_location                      2415
neighbourhood_group_cleansed          0
latitude                              0
longitude                             0
room_type                             0
accommodates                          0
bathrooms_text                        2
bedrooms                          13327
beds                                208
price                              3953
minimum_nights                        0
maximum_nights                        0
availability_365                      0
license                            5091
instant_bookable                      0
calculated_host_listings_count        0
dtype: int64

The bedrooms col. has no values, maybe is due to a scraping error. This should not inluence the analysis since I should be able to retrieve the data in the other files. 

In [767]:
# Keep only numerical values for bathrooms_text
listing2 = listing2.assign(bathrooms=listing2['bathrooms_text'].str.extract(r'(\d+[\.,]?\d*)'))

In [769]:
# Change data type for bathrooms
listing2['bathrooms'] = listing2['bathrooms'].astype(float)

In [771]:
# Drop bathrooms_text col. 
listing2.drop(columns='bathrooms_text', axis=1, inplace=True)

In [481]:
# Look for missing values in bathrooms
listing2['bathrooms'].isnull().sum()

46

In [773]:
# Replace the $ sign in price
listing2['price'] = listing2['price'].str.replace('$', '', regex=False).str.replace(',','', regex=False).str.strip()

In [775]:
# Change data type for price
listing2['price'] = listing2['price'].astype(float)

In [85]:
# Summary Statistics
listing2.describe()

id       host_id      latitude     longitude  accommodates  \
count  1.332700e+04  1.332700e+04  13327.000000  13327.000000  13327.000000   
mean   2.726391e+17  1.481982e+08     52.509764     13.402477      2.934794   
std    4.000348e+17  1.686400e+08      0.033481      0.066762      1.864588   
min    3.176000e+03  1.581000e+03     52.369040     13.107580      1.000000   
25%    1.723423e+07  1.362112e+07     52.490371     13.364285      2.000000   
50%    3.916137e+07  6.108240e+07     52.509630     13.411440      2.000000   
75%    6.968867e+17  2.577263e+08     52.532395     13.438163      4.000000   
max    1.049081e+18  5.510794e+08     52.656110     13.721390     16.000000   

       bedrooms          beds         price  minimum_nights  maximum_nights  \
count       0.0  13119.000000   9374.000000    13327.000000    13327.000000   
mean        NaN      1.842823    124.517068       42.797329      620.284235   
std         NaN      1.466189    383.751953       53.963528      488.014129   
min         NaN      1.000000      9.000000        1.000000        1.000000   
25%         NaN      1.000000     62.000000        2.000000      120.000000   
50%         NaN      1.000000     91.000000        6.000000      365.000000   
75%         NaN      2.000000    139.000000       92.000000     1125.000000   
max         NaN     29.000000  24999.000000     1100.000000     9999.000000   

       availability_365  calculated_host_listings_count     bathrooms  
count      13327.000000                    13327.000000  13281.000000  
mean         120.597809                        9.276131      1.138920  
std          135.093951                       31.789247      0.406201  
min            0.000000                        1.000000      0.000000  
25%            0.000000                        1.000000      1.000000  
50%           58.000000                        1.000000      1.000000  
75%          257.000000                        3.000000      1.000000  
max          365.000000                      231.000000      8.500000

In [777]:
# Calculate median for price
listing2['price'].median()

91.0

In [779]:
# Imput mean for 'beds', 'bathrooms', median for 'price'
listing2['beds'] = listing2['beds'].fillna(value=2)
listing2['bathrooms'] = listing2['bathrooms'].fillna(value=1)
listing2['price'] = listing2['price'].fillna(value=91)

In [781]:
# Add boolean col. for license
listing2['license_bool'] = listing2['license'].notnull()

In [783]:
# Drop license col. 
listing2.drop(columns='license', axis=1, inplace=True)

In [95]:
# Check data types
listing2.dtypes

id                                  int64
last_scraped                       object
host_id                             int64
host_location                      object
neighbourhood_group_cleansed       object
latitude                          float64
longitude                         float64
room_type                          object
accommodates                        int64
bedrooms                          float64
beds                              float64
price                             float64
minimum_nights                      int64
maximum_nights                      int64
availability_365                    int64
instant_bookable                   object
calculated_host_listings_count      int64
bathrooms                         float64
license_bool                         bool
dtype: object

In [785]:
# Change data type for 'id', 'host_id', 'price', 'host_is_superhost', 'instant_bookable'
listing2 = listing2.astype({'id':str, 'host_id':str, 'instant_bookable':bool})

In [787]:
# Apply dictionary to change cols names 
listing2.rename(columns=new_name, inplace=True)

In [101]:
# Check the changes
listing2.columns

Index(['id', 'last_scraped', 'host_id', 'host_location', 'neighbourhood',
       'latitude', 'longitude', 'room_type', 'accommodates', 'bedrooms',
       'beds', 'price', 'minimum_nights', 'maximum_nights', 'availability_365',
       'instant_bookable', 'host_listings', 'bathrooms', 'license'],
      dtype='object')

##### Lst3

In [181]:
#Lst3 size
lst3.shape

(13362, 75)

In [186]:
# Lst3 columns
lst3.columns

Index(['id', 'listing_url', 'scrape_id', 'last_scraped', 'source', 'name',
       'description', 'neighborhood_overview', 'picture_url', 'host_id',
       'host_url', 'host_name', 'host_since', 'host_location', 'host_about',
       'host_response_time', 'host_response_rate', 'host_acceptance_rate',
       'host_is_superhost', 'host_thumbnail_url', 'host_picture_url',
       'host_neighbourhood', 'host_listings_count',
       'host_total_listings_count', 'host_verifications',
       'host_has_profile_pic', 'host_identity_verified', 'neighbourhood',
       'neighbourhood_cleansed', 'neighbourhood_group_cleansed', 'latitude',
       'longitude', 'property_type', 'room_type', 'accommodates', 'bathrooms',
       'bathrooms_text', 'bedrooms', 'beds', 'amenities', 'price',
       'minimum_nights', 'maximum_nights', 'minimum_minimum_nights',
       'maximum_minimum_nights', 'minimum_maximum_nights',
       'maximum_maximum_nights', 'minimum_nights_avg_ntm',
       'maximum_nights_avg_ntm', 'ca

In [789]:
# Select only relevant variables
listing3 = lst3[['id', 'last_scraped', 'host_id','host_location', 'neighbourhood_group_cleansed', 'latitude','longitude', 'room_type', 'accommodates','bathrooms_text', 'bedrooms', 'beds', 'price','minimum_nights', 'maximum_nights', 'availability_365', 'license', 'instant_bookable','calculated_host_listings_count']]

In [501]:
# Look for duplicates
dup = listing3.duplicated()
dup.sum()

0

In [503]:
# Look for missing values
listing3.isnull().sum()

id                                   0
last_scraped                         0
host_id                              0
host_location                     2526
neighbourhood_group_cleansed         0
latitude                             0
longitude                            0
room_type                            0
accommodates                         0
bathrooms_text                       5
bedrooms                          2141
beds                              4970
price                             4962
minimum_nights                       0
maximum_nights                       0
availability_365                     0
license                           4965
instant_bookable                     0
calculated_host_listings_count       0
dtype: int64

In [791]:
# Keep only numerical values for bathrooms_text
listing3 = listing3.assign(bathrooms=listing3['bathrooms_text'].str.extract(r'(\d+[\.,]?\d*)')) 

In [793]:
# Change data type for bathroms
listing3['bathrooms'] = listing3['bathrooms'].astype(float)

In [795]:
# Drop bathrooms_text col. 
listing3.drop(columns='bathrooms_text', axis=1, inplace=True)

In [511]:
# Look for missing values in bathrooms
listing3['bathrooms'].isnull().sum()

59

In [797]:
# Replace the $ sign in price
listing3['price'] = listing3['price'].str.replace('$', '', regex=False).str.replace(',', '', regex=False).str.strip()

In [799]:
# Change data type for price
listing3['price'] = listing3['price'].astype(float)

In [121]:
#Summary Statistics
listing3.describe()

id       host_id      latitude     longitude  accommodates  \
count  1.336200e+04  1.336200e+04  13362.000000  13362.000000  13362.000000   
mean   3.102020e+17  1.538724e+08     52.509676     13.402961      2.964826   
std    4.287225e+17  1.747436e+08      0.033598      0.066296      1.879211   
min    3.176000e+03  1.581000e+03     52.369040     13.118150      1.000000   
25%    1.803007e+07  1.391377e+07     52.490260     13.364565      2.000000   
50%    4.060840e+07  6.387887e+07     52.509651     13.411290      2.000000   
75%    7.679724e+17  2.667629e+08     52.532229     13.438257      4.000000   
max    1.118938e+18  5.681830e+08     52.656110     13.721390     16.000000   

           bedrooms         beds        price  minimum_nights  maximum_nights  \
count  11221.000000  8392.000000  8400.000000    13362.000000    13362.000000   
mean       1.344711     1.944471   128.425714       42.139725      599.825924   
std        0.832744     1.625830   124.398528       53.564546      486.285018   
min        0.000000     0.000000     9.000000        1.000000        1.000000   
25%        1.000000     1.000000    67.000000        2.000000       99.250000   
50%        1.000000     1.000000    99.000000        5.000000      365.000000   
75%        2.000000     2.000000   150.000000       92.000000     1125.000000   
max       15.000000    26.000000  3429.000000     1100.000000     9999.000000   

       availability_365  calculated_host_listings_count     bathrooms  
count      13362.000000                    13362.000000  13303.000000  
mean         122.829666                       10.123485      1.138916  
std          128.440536                       36.114710      0.409286  
min            0.000000                        1.000000      0.000000  
25%            0.000000                        1.000000      1.000000  
50%           81.000000                        1.000000      1.000000  
75%          241.000000                        4.000000      1.000000  
max          365.000000                      255.000000      8.500000

In [801]:
# Calculate median for price
listing3['price'].median()

99.0

In [803]:
# Imput mean for 'bedrooms', 'beds', 'bathrooms', median for 'price'
listing3['bedrooms'] = listing3['bedrooms'].fillna(value=1)
listing3['beds'] = listing3['beds'].fillna(value=2)
listing3['bathrooms'] = listing3['bathrooms'].fillna(value=1)
listing3['price'] = listing3['price'].fillna(value=99)

In [805]:
# Add boolean col. for license
listing3['license_bool'] = listing3['license'].notnull()

In [807]:
# Drop license col. 
listing3.drop(columns='license', axis=1, inplace=True)

In [129]:
# Check data types
listing3.dtypes

id                                  int64
last_scraped                       object
host_id                             int64
host_location                      object
neighbourhood_group_cleansed       object
latitude                          float64
longitude                         float64
room_type                          object
accommodates                        int64
bedrooms                          float64
beds                              float64
price                             float64
minimum_nights                      int64
maximum_nights                      int64
availability_365                    int64
license                            object
instant_bookable                   object
calculated_host_listings_count      int64
bathrooms                         float64
license_bool                         bool
dtype: object

In [809]:
# Change data type for 'id', 'host_id', 'host_is_superhost', 'instant_bookable'
listing3 = listing3.astype({'id':str, 'host_id':str, 'instant_bookable':bool})

In [811]:
# Apply dictionary to change cols names
listing3.rename(columns=new_name, inplace=True)

In [135]:
# Check the changes
listing3.columns

Index(['id', 'last_scraped', 'host_id', 'host_location', 'neighbourhood',
       'latitude', 'longitude', 'room_type', 'accommodates', 'bedrooms',
       'beds', 'price', 'minimum_nights', 'maximum_nights', 'availability_365',
       'license', 'instant_bookable', 'host_listings', 'bathrooms', 'license'],
      dtype='object')

##### Lst4

In [244]:
# Lst4 size
lst4.shape

(13759, 75)

In [247]:
# Lst4 columns names
lst4.columns

Index(['id', 'listing_url', 'scrape_id', 'last_scraped', 'source', 'name',
       'description', 'neighborhood_overview', 'picture_url', 'host_id',
       'host_url', 'host_name', 'host_since', 'host_location', 'host_about',
       'host_response_time', 'host_response_rate', 'host_acceptance_rate',
       'host_is_superhost', 'host_thumbnail_url', 'host_picture_url',
       'host_neighbourhood', 'host_listings_count',
       'host_total_listings_count', 'host_verifications',
       'host_has_profile_pic', 'host_identity_verified', 'neighbourhood',
       'neighbourhood_cleansed', 'neighbourhood_group_cleansed', 'latitude',
       'longitude', 'property_type', 'room_type', 'accommodates', 'bathrooms',
       'bathrooms_text', 'bedrooms', 'beds', 'amenities', 'price',
       'minimum_nights', 'maximum_nights', 'minimum_minimum_nights',
       'maximum_minimum_nights', 'minimum_maximum_nights',
       'maximum_maximum_nights', 'minimum_nights_avg_ntm',
       'maximum_nights_avg_ntm', 'ca

In [813]:
# Select only relevant variables
listing4 = lst4[['id', 'last_scraped', 'host_id','host_location', 'neighbourhood_group_cleansed', 'latitude','longitude', 'room_type', 'accommodates','bathrooms_text', 'bedrooms', 'beds', 'price','minimum_nights', 'maximum_nights', 'availability_365', 'license', 'instant_bookable','calculated_host_listings_count']]

In [183]:
# Look for duplicates
dup = listing4.duplicated()
dup.sum()

0

In [185]:
# Look for missing values
listing4.isnull().sum()

id                                   0
last_scraped                         0
host_id                              0
host_location                     2696
neighbourhood_group_cleansed         0
latitude                             0
longitude                            0
room_type                            0
accommodates                         0
bathrooms_text                       5
bedrooms                          2057
beds                              5001
price                             4938
minimum_nights                       0
maximum_nights                       0
availability_365                     0
license                           4956
instant_bookable                     0
calculated_host_listings_count       0
dtype: int64

In [815]:
# Keep only numerical values for bathrooms_text
listing4 = listing4.assign(bathrooms=listing4['bathrooms_text'].str.extract(r'(\d+[\.,]?\d*)'))

In [817]:
# Change data type for bathrooms
listing4['bathrooms'] = listing4['bathrooms'].astype(float)

In [819]:
# Drop bathrooms_text
listing4.drop(columns='bathrooms_text', axis=1, inplace=True)

In [537]:
# Look for missing values in bathrooms
listing4['bathrooms'].isnull().sum()

67

In [821]:
# Replace the $ sign in price
listing4['price'] = listing4['price'].str.replace('$', '', regex=False).str.replace(',', '', regex=False).str.strip()

In [823]:
# Change data type for price
listing4['price'] = listing4['price'].astype(float)

In [201]:
# Summary Statistics
listing4.describe()

id       host_id      latitude     longitude  accommodates  \
count  1.375900e+04  1.375900e+04  13759.000000  13759.000000  13759.000000   
mean   3.529798e+17  1.618469e+08     52.509342     13.402723      2.993968   
std    4.588343e+17  1.822188e+08      0.033789      0.067338      1.896326   
min    3.176000e+03  1.581000e+03     52.340190     13.118150      1.000000   
25%    1.884756e+07  1.447609e+07     52.490077     13.362765      2.000000   
50%    4.234426e+07  6.789678e+07     52.509220     13.411275      2.000000   
75%    8.461250e+17  2.785392e+08     52.532147     13.438580      4.000000   
max    1.184030e+18  5.846847e+08     52.656110     13.721390     16.000000   

           bedrooms         beds         price  minimum_nights  \
count  11702.000000  8758.000000   8821.000000    13759.000000   
mean       1.346180     1.861612    154.840381       40.905952   
std        0.829784     1.683820    296.293035       52.400853   
min        0.000000     0.000000      9.000000        1.000000   
25%        1.000000     1.000000     74.000000        2.000000   
50%        1.000000     1.000000    114.000000        5.000000   
75%        2.000000     2.000000    186.000000       92.000000   
max       15.000000    26.000000  24277.000000     1000.000000   

       maximum_nights  availability_365  calculated_host_listings_count  \
count    13759.000000      13759.000000                    13759.000000   
mean       583.216949        125.307799                       11.022531   
std        483.065031        129.410364                       40.753865   
min          1.000000          0.000000                        1.000000   
25%         92.000000          0.000000                        1.000000   
50%        365.000000         82.000000                        1.000000   
75%       1125.000000        252.000000                        4.000000   
max       9999.000000        365.000000                      281.000000   

          bathrooms  
count  13692.000000  
mean       1.141177  
std        0.428566  
min        0.000000  
25%        1.000000  
50%        1.000000  
75%        1.000000  
max       15.000000

In [543]:
# Calculate median for price
listing4['price'].median()

114.0

In [825]:
# Imput mean for 'bedrooms', beds, 'bathrooms', median for 'price'
listing4['bedrooms'] = listing4['bedrooms'].fillna(value=1)
listing4['beds'] = listing4['beds'].fillna(value=2)
listing4['bathrooms'] = listing4['bathrooms'].fillna(value=1)
listing4['price'] = listing4['price'].fillna(value=114)

In [827]:
# Add boolean col. for license
listing4['license_bool'] = listing4['license'].notnull()

In [829]:
# Drop license col. 
listing4.drop(columns='license', axis=1, inplace=True)

In [211]:
# Check data types
listing4.dtypes

id                                  int64
last_scraped                       object
host_id                             int64
host_location                      object
neighbourhood_group_cleansed       object
latitude                          float64
longitude                         float64
room_type                          object
accommodates                        int64
bedrooms                          float64
beds                              float64
price                             float64
minimum_nights                      int64
maximum_nights                      int64
availability_365                    int64
instant_bookable                   object
calculated_host_listings_count      int64
bathrooms                         float64
license_bool                         bool
dtype: object

In [831]:
# Change data type for 'id', 'host_id', 'host_is_superhost', 'instant_bookable'
listing4 = listing4.astype({'id':str, 'host_id':str, 'instant_bookable':bool})

In [833]:
# Apply dictionary to change cols names
listing4.rename(columns=new_name, inplace=True)

In [217]:
# Check changes
listing4.columns

Index(['id', 'last_scraped', 'host_id', 'host_location', 'neighbourhood',
       'latitude', 'longitude', 'room_type', 'accommodates', 'bedrooms',
       'beds', 'price', 'minimum_nights', 'maximum_nights', 'availability_365',
       'instant_bookable', 'host_listings', 'bathrooms', 'license'],
      dtype='object')

## Handling Outliers and Inconsistencies

##### Max price of 24.999 for listing 1 and 2 and price of 24.277 for listing 4 are too high even for luxury listings

In [835]:
# Calculate mean and st. deviation for listing1
mean = listing1['price'].mean()
std = listing1['price'].std()

In [837]:
# Z-score
listing1['z_score'] = (listing1['price'] - mean) / std

In [839]:
# Threshold
threshold = 3

In [841]:
# Outliers
outliers = listing1[np.abs(listing1['z_score']) > threshold]

In [563]:
# Printing id and price of the outliers for investigation
to_print = ['id', 'price']

In [323]:
out = outliers[to_print]

In [326]:
print(out)

                       id    price
713               2860420   2500.0
1693              7807397   3429.0
5401             29022451   1497.0
5402             29022524   1691.0
6930             39877586   3000.0
7282             41787739   2310.0
7298             41897771  24999.0
7312             41923918  24999.0
7314             41924747  24999.0
9050             53712999   1338.0
9065             53733309   1424.0
9830   648389751444769673   2088.0
9945   662966098610376381   1900.0
10338  706232116352290861   1800.0
10378  712058799866210922   1425.0
12066  909318377890699731   1373.0
12069  909527058663796716  10000.0


Records: 7298, 7312, 7314, 12069 are not listed anymore on AirBnb and the price per night is too hight, even for luxury hotels or apartments. 

In [843]:
# Calculate mean and st. deviation for listing2
mean = listing2['price'].mean()
std = listing2['price'].std()

In [845]:
# Z-score
listing2['z_score'] = (listing2['price'] - mean) / std

In [847]:
# Outliers
outliers = listing2[np.abs(listing2['z_score']) > threshold]

In [343]:
# Printing id and price of the outliers for investigation
to_print = ['id', 'price']

In [343]:
out = outliers[to_print]

In [346]:
print(out)

                       id    price
453               1342115   1260.0
727               2860420   2500.0
1715              7807397   3429.0
3255             16799799   1375.0
5287             29022524   1594.0
5396             29509865   1358.0
5397             29509869   1659.0
6248             36165198   1611.0
6621             38502596   1200.0
6764             39877586   3000.0
7016             41224162   1457.0
7080             41897771  24999.0
7107             41924747  24999.0
7183             41787739   2310.0
8035             47742881   1146.0
9596   648389751444769673   1856.0
9673   662966098610376381   1900.0
12239  965294059855869245   2500.0
12606  993210282733742154   1800.0


Records: 7080, 7107 are not listed anymore on AirBnb and the price per night is too hight, even for luxury hotels or apartments.

In [849]:
# Calculate mean and st. deviation for listing4
mean = listing4['price'].mean()
std = listing4['price'].std()

In [851]:
# Z-score
listing4['z_score'] = (listing4['price'] - mean) / std

In [853]:
# Outliers
outliers = listing4[np.abs(listing4['z_score']) > threshold]

In [243]:
# Change display options for web scraping
pd.set_option('display.max_rows', None)

In [577]:
# Printing id and price of the outliers for investigation
to_print = ['id', 'price']

In [366]:
out = outliers[to_print]

In [369]:
print(out)

                        id    price
422                1342115   1260.0
682                2860420   2500.0
1610               7807397   3429.0
1880               9268629   1058.0
2569              13060403    950.0
2698              13868134    925.0
3024              16245894    985.0
3126              16799799   1250.0
3260              17786473    885.0
3491              19153150   1822.0
3637              19961235   1798.0
4055              22075653    928.0
4216              22966047   1287.0
5042              29022451   1014.0
5043              29022524   1178.0
5343              31278404  24277.0
5347              31297828   1129.0
6060              36871553    959.0
6258              38502596   1200.0
6446              39877586   3000.0
6587              40650136    860.0
6990              42803135    999.0
6991              42804250    999.0
6992              42805419    999.0
6993              42805553    999.0
6994              42805923    999.0
7599              47742881  

Record 5343 is not listed anymore on AirBnb and the price per night is too hight, even for luxury hotels or apartments. 

In [855]:
# Drop outliers that are not listed anymore
listing1 = listing1.drop([7298, 7312, 7314, 12069])
listing2 = listing2.drop([7080, 7107])
listing4 = listing4.drop(5343)

In [857]:
# Drop z_score cols. 
listing1.drop('z_score', axis=1, inplace=True)
listing2.drop('z_score', axis=1, inplace=True)
listing4.drop('z_score', axis=1, inplace=True)

##### Minimum Beds 0 in listing3 and listing4 is too low

In [859]:
# Function to update beds based on accommodates
def update_beds(accommodates_value):
    return max(1, accommodates_value // 2)

In [861]:
# Apply function to listing3
listing3['beds'] = listing3['accommodates'].apply(update_beds)

In [863]:
# Apply function to listing4
listing4['beds'] = listing4['accommodates'].apply(update_beds)

In [865]:
# Check min value 
listing3['beds'].min()

1

In [867]:
# Check min value 
listing4['beds'].min()

1

##### Max minimum_nights 1100 seems too high for a short rental

##### listing1

In [869]:
# Calculate mean and st. deviation 
mean = listing1['minimum_nights'].mean()
std = listing1['minimum_nights'].std()

In [871]:
# Z-score
listing1['z_score'] = (listing1['minimum_nights'] - mean) / std

In [873]:
# Outliers
outliers = listing1[np.abs(listing1['z_score']) > threshold]

In [595]:
# Printing 'id', 'minimum_nights', 'availability_365' for further investigation
to_print = ['id', 'minimum_nights', 'availability_365']

In [415]:
out = outliers[to_print]

In [417]:
print(out)

                       id  minimum_nights  availability_365
55                 174547             365                 0
56                 174560             365                 0
57                 174576             365                 0
58                 174579             365                 0
231                584757            1000               365
290                769334             360               363
371               1106372            1100                 0
426               1318989             300               365
815               3584113             800               365
911               4003140             500               365
973               4314353             355                 0
1028              4584283             365               107
1411              6670861             700               365
1424              6704144            1000               365
1648              7682402            1000               365
2025              9484672             21

All the outliers listing with 0 availability in the next 365 days have also no reviews online or very old ones, so most likely are not actually available for any tipe of renting.

In [875]:
# To delete
delete = outliers.loc[outliers['availability_365'] == 0]

In [877]:
delete.shape

(11, 20)

In [879]:
# Drop the values
listing1 = listing1.drop(delete.index)

In [881]:
listing1.shape

(13119, 20)

##### listing2

In [883]:
# Calculate mean and st. deviation
mean = listing2['minimum_nights'].mean()
std = listing2['minimum_nights'].std()

In [885]:
# Z-score
listing2['z_score'] = (listing2['minimum_nights'] - mean) / std

In [887]:
# Outliers
outliers = listing2[np.abs(listing2['z_score']) > threshold]

In [889]:
# To delete
delete = outliers.loc[outliers['availability_365'] == 0]

In [891]:
delete.shape

(13, 20)

In [893]:
# Drop the values
listing2 = listing2.drop(delete.index)

In [895]:
listing2.shape

(13312, 20)

##### listing3

In [897]:
# Calculate mean and st. deviation for listing3
mean = listing3['minimum_nights'].mean()
std = listing3['minimum_nights'].std()

In [899]:
# Z-score
listing3['z_score'] = (listing3['minimum_nights'] - mean) / std

In [901]:
# Outliers
outliers = listing3[np.abs(listing3['z_score']) > threshold]

In [903]:
# to delete
delete = outliers.loc[outliers['availability_365'] == 0]

In [905]:
delete.shape

(12, 20)

In [907]:
# Drop the values 
listing3 = listing3.drop(delete.index)

In [909]:
listing3.shape

(13350, 20)

In [315]:
listing3.columns

Index(['id', 'last_scraped', 'host_id', 'host_location', 'neighbourhood',
       'latitude', 'longitude', 'room_type', 'accommodates', 'bedrooms',
       'beds', 'price', 'minimum_nights', 'maximum_nights', 'availability_365',
       'license', 'instant_bookable', 'host_listings', 'bathrooms', 'license',
       'z_score'],
      dtype='object')

##### listing4

In [911]:
# Calculate mean and st. deviation for listing4
mean = listing4['minimum_nights'].mean()
std = listing4['minimum_nights'].std()

In [913]:
# Z-score
listing4['z_score'] = (listing4['minimum_nights'] - mean) /std

In [915]:
# Outliers
outiers = listing4[np.abs(listing4['z_score']) > threshold]

In [917]:
# to delete
delete = outliers.loc[outliers['availability_365'] == 0]

In [919]:
delete.shape

(12, 20)

In [921]:
# Drop the values 
listing4 = listing4.drop(delete.index)

In [923]:
listing4.shape

(13746, 20)

In [925]:
# Drop z_score cols. 
listing1.drop(columns='z_score', axis=1, inplace=True)
listing2.drop(columns='z_score', axis=1, inplace=True)
listing3.drop(columns='z_score', axis=1, inplace=True)
listing4.drop(columns='z_score', axis=1, inplace=True)

##### Min 0 bathrooms needs further investigation

In [927]:
# Look for 0 bathrooms in listing1
baths = listing1.loc[listing1['bathrooms'] == 0]

In [508]:
to_print = ['id', 'bathrooms']

In [511]:
bath = baths[to_print]

In [514]:
print(bath)

                       id  bathrooms
137                265389        0.0
138                265394        0.0
139                265402        0.0
140                265408        0.0
338                926497        0.0
733               3088705        0.0
737               3108376        0.0
738               3108596        0.0
739               3108686        0.0
740               3109139        0.0
1076              4816644        0.0
1267              5875924        0.0
1348              6322584        0.0
4125             20990489        0.0
4674             24057633        0.0
5052             26839501        0.0
5064             26950491        0.0
5638             30759237        0.0
5795             31817631        0.0
6026             33567229        0.0
6718             38502596        0.0
7187             41130331        0.0
7196             41202289        0.0
7575             43055810        0.0
7853             44590435        0.0
7949             45464162        0.0
8

After checking the listings on airbnb, I can confirm that the value 0 means that it's a shared communal bathroom. 

##### Checking Max Value for Host Listings

##### listing1

In [929]:
# Calculate mean and st. deviation
mean = listing1['host_listings'].mean()
std = listing1['host_listings'].std()

In [931]:
# Z-score
listing1['z_score'] = (listing1['host_listings'] - mean) / std

In [933]:
# Outliers
outliers = listing1[np.abs(listing1['z_score']) > threshold]

In [657]:
outliers['host_id'].value_counts()

434485440    187
367035318    114
480052472    113
Name: host_id, dtype: int64

Host_id 434485440 is a short-term apartments rental company. 367035318 and 480052472 are not available online anymore, but they had probably a similar purpose. 

##### listing2

In [935]:
# Calculate mean and st. deviation
mean = listing2['host_listings'].mean()
std = listing2['host_listings'].std()

In [937]:
# Z-score
listing2['z_score'] = (listing2['host_listings'] - mean) / std

In [939]:
# Outliers
outliers = listing2[np.abs(listing2['z_score']) > threshold]

In [665]:
outliers['host_id'].value_counts()

434485440    231
Name: host_id, dtype: int64

##### listing3

In [941]:
# Calculate mean and st. deviation
mean = listing3['host_listings'].mean()
std = listing3['host_listings'].std()

In [943]:
# Z-score
listing3['z_score'] = (listing3['host_listings'] - mean) / std

In [945]:
# Outliers
outliers = listing3[listing3['z_score'] > threshold]

In [673]:
outliers['host_id'].value_counts()

434485440    255
Name: host_id, dtype: int64

##### listing4

In [947]:
# Calculate mean and st. deviation
mean = listing4['host_listings'].mean()
std = listing4['host_listings'].std()

In [949]:
# Z-score
listing4['z_score'] = (listing4['host_listings'] - mean) / std

In [951]:
# Outliers
outliers = listing4[listing4['z_score'] > threshold]

In [681]:
outliers['host_id'].value_counts()

434485440    281
Name: host_id, dtype: int64

In [953]:
# Drop z_score cols. 
listing1.drop(columns='z_score', axis=1, inplace=True)
listing2.drop(columns='z_score', axis=1, inplace=True)
listing3.drop(columns='z_score', axis=1, inplace=True)
listing4.drop(columns='z_score', axis=1, inplace=True)

### Merge DFs to fill listing2 bedrooms missing values

In [955]:
# Merge listing1 and listing2
merged1 = listing2.merge(listing1, on='id', how='left')

In [957]:
merged1['bedrooms_x'] = merged1['bedrooms_x'].fillna(merged1['bedrooms_y'])

In [959]:
# Look for null values after merging
merged1['bedrooms_x'].isnull().sum()

1525

In [961]:
merged1.columns

Index(['id', 'last_scraped_x', 'host_id_x', 'host_location_x',
       'neighbourhood_x', 'latitude_x', 'longitude_x', 'room_type_x',
       'accommodates_x', 'bedrooms_x', 'beds_x', 'price_x', 'minimum_nights_x',
       'maximum_nights_x', 'availability_365_x', 'instant_bookable_x',
       'host_listings_x', 'bathrooms_x', 'license_x', 'last_scraped_y',
       'host_id_y', 'host_location_y', 'neighbourhood_y', 'latitude_y',
       'longitude_y', 'room_type_y', 'accommodates_y', 'bedrooms_y', 'beds_y',
       'price_y', 'minimum_nights_y', 'maximum_nights_y', 'availability_365_y',
       'instant_bookable_y', 'host_listings_y', 'bathrooms_y', 'license_y'],
      dtype='object')

In [963]:
# Drop unnecessary columns
merged1.drop(columns=['last_scraped_y','host_id_y', 'host_location_y','neighbourhood_y', 'latitude_y','longitude_y', 'room_type_y', 'accommodates_y', 'bedrooms_y', 'beds_y','price_y', 'minimum_nights_y', 'maximum_nights_y', 'availability_365_y','instant_bookable_y', 'host_listings_y', 'bathrooms_y', 'license_y'], axis=1, inplace=True)

In [965]:
# Dictionary
dict = {'last_scraped_x':'last_scraped', 'host_id_x':'host_id', 'host_location_x':'host_location','neighbourhood_x':'neighbourhood',
       'latitude_x':'latitude', 'longitude_x':'longitude', 'room_type_x':'room_type', 'accommodates_x':'accommodates',
       'bedrooms_x':'bedrooms', 'beds_x':'beds', 'price_x':'price', 'minimum_nights_x':'minimum_nights',
       'maximum_nights_x':'maximum_nights', 'availability_365_x':'availability_365', 'instant_bookable_x':'instant_bookable',
       'host_listings_x':'host_listings', 'bathrooms_x':'bathrooms', 'license_x':'license'}

In [967]:
# Rename columns
merged1.rename(columns=dict, inplace=True)

In [969]:
# Calculate mean for bedrooms
merged1['bedrooms'].mean()

1.312378043607364

In [971]:
# Imput mean for bedrooms missing values
merged1['bedrooms'] = merged1['bedrooms'].fillna(value=1)

#### GeoJson file

In [973]:
# Load GeoJson
with open(geo, 'r') as f:
    geojson_data = json.load(f)

In [975]:
# Extract names/identifiers from the 'properties'
geojson_names = [feature['properties'].get('Gemeinde_name') for feature in geojson_data['features']]

In [622]:
print(geojson_names)

['Reinickendorf', 'Charlottenburg-Wilmersdorf', 'Treptow-Köpenick', 'Pankow', 'Neukölln', 'Lichtenberg', 'Marzahn-Hellersdorf', 'Spandau', 'Steglitz-Zehlendorf', 'Mitte', 'Friedrichshain-Kreuzberg', 'Tempelhof-Schöneberg']


In [977]:
# Check the neighbourhoods variables in listing1 to match the values
listing1['neighbourhood'].unique()

array(['Pankow', 'Friedrichshain-Kreuzberg', 'Neukölln', 'Mitte',
       'Charlottenburg-Wilm.', 'Tempelhof - Schöneberg',
       'Treptow - Köpenick', 'Steglitz - Zehlendorf', 'Lichtenberg',
       'Marzahn - Hellersdorf', 'Spandau', 'Reinickendorf'], dtype=object)

In [979]:
# Dictionary
dict = {'Charlottenburg-Wilm.':'Charlottenburg-Wilmersdorf', 'Tempelhof - Schöneberg':'Tempelhof-Schöneberg', 'Treptow - Köpenick':'Treptow-Köpenick', 'Steglitz - Zehlendorf':'Steglitz-Zehlendorf', 'Marzahn - Hellersdorf':'Marzahn-Hellersdorf'}

In [981]:
# Apply dictionary
listing1['neighbourhood'] = listing1['neighbourhood'].replace(dict)
merged1['neighbourhood'] = merged1['neighbourhood'].replace(dict)
listing3['neighbourhood'] = listing3['neighbourhood'].replace(dict)
listing4['neighbourhood'] = listing4['neighbourhood'].replace(dict)

## Export Files

In [983]:
listing1.to_pickle(os.path.join(path, 'Prepared', 'listing1_clean.pkl'))

In [985]:
merged1.to_pickle(os.path.join(path, 'Prepared', 'listing2_clean.pkl'))

In [987]:
listing3.to_pickle(os.path.join(path, 'Prepared', 'listing3_clean.pkl'))

In [989]:
listing4.to_pickle(os.path.join(path, 'Prepared', 'listing4_clean.pkl'))

## Concatenate Dataframes

In [991]:
# Concatenate dataframes
listing_concat = pd.concat([listing1, merged1, listing3, listing4], ignore_index=True)

In [993]:
listing_concat.shape

(53527, 19)

In [995]:
listing_concat.columns

Index(['id', 'last_scraped', 'host_id', 'host_location', 'neighbourhood',
       'latitude', 'longitude', 'room_type', 'accommodates', 'bedrooms',
       'beds', 'price', 'minimum_nights', 'maximum_nights', 'availability_365',
       'instant_bookable', 'host_listings', 'bathrooms', 'license'],
      dtype='object')

In [997]:
# Summary Statistics
listing_concat.describe()

latitude     longitude  accommodates      bedrooms          beds  \
count  53527.000000  53527.000000  53527.000000  53527.000000  53527.000000   
mean      52.509679     13.402454      2.953239      1.291049      1.647393   
std        0.033590      0.067064      1.874246      0.726829      1.198251   
min       52.340190     13.107580      1.000000      0.000000      1.000000   
25%       52.490265     13.363385      2.000000      1.000000      1.000000   
50%       52.509570     13.411290      2.000000      1.000000      1.000000   
75%       52.532370     13.438265      4.000000      1.000000      2.000000   
max       52.656110     13.721390     16.000000     15.000000     29.000000   

              price  minimum_nights  maximum_nights  availability_365  \
count  53527.000000    53527.000000    53527.000000      53527.000000   
mean     120.682272       41.983149      610.596577        123.936499   
std      111.890354       51.370580      488.008174        132.092173   
min        8.000000        1.000000        1.000000          0.000000   
25%       72.000000        2.000000      100.000000          0.000000   
50%       99.000000        6.000000      365.000000         74.000000   
75%      128.000000       92.000000     1125.000000        255.000000   
max     3429.000000     1000.000000     9999.000000        365.000000   

       host_listings     bathrooms  
count   53527.000000  53527.000000  
mean        9.936574      1.137725  
std        34.560510      0.410125  
min         1.000000      0.000000  
25%         1.000000      1.000000  
50%         1.000000      1.000000  
75%         3.000000      1.000000  
max       281.000000     15.000000

In [999]:
# Export concatenated file
listing_concat.to_pickle(os.path.join(path, 'Prepared', 'list_concat.pkl'))